kornia/augmentation/tormentor_minimal/spatial_augmentations.py

```diff
@@ -0,0 +1,134 @@
+from base_augmentation import SpatialImageAugmentation, aug_parameters, aug_statenames
+#from .backgrounds import *
+
+import torch
+import torch.nn.functional as F
+import kornia
+import math
+
+
+# Simple example of using the augmentatio n API
+@aug_statenames("rotate_radians")
+@aug_parameters(min_angle=-math.pi, max_angle=math.pi)
+class Rotate(SpatialImageAugmentation):
+    def forward_batch(self, tensor_image):
+        self.rotate_radians = (torch.rand(tensor_image.size(0)) + self.min_angle)*(self.max_angle-self.min_angle)
+        rotate_degrees = (self.rotate_radians * 180) / math.pi
+        print(rotate_degrees)
+        return kornia.geometry.rotate(tensor_image, rotate_degrees)
+
+    def forward_sample(self, tensor_image):
+        tensor_image = tensor_image.unsqueeze(dim=0)
+        self.rotate_radians = (torch.rand(1) + self.min_angle)*(self.max_angle-self.min_angle)
+        rotate_degrees = (self.rotate_radians * 180) / math.pi
+        print(rotate_degrees)
+        return kornia.geometry.rotate(tensor_image, rotate_degrees)[0, :, :, :]
+
+
+@aug_parameters(min_scale=.5, max_scale=2.0, joint=False)
```

Comment:
I personally do not like the joint parameter. I think we can use max_scale=2.0 to represent joint, max_scale=[2.0, 1.0] to represent non-joint.

Reply:
That is definetly more elegant. A proper typeunion for valid random ranges should be broadly used IMO.

Reply:
tuple might be more suitable in this context

Reply:
I my new draft (), I think a light wrapper around torch.distributions would be even more appropriate. Simething like Uniform(.5,2.0).